# Author: dtrip <d@nrx.co>
# Date: 2015.10.28 10:26UTC
#
# Description:
# Patch file for rtl8812au drivers to work
# with Amped wireless UA230A on raspberry pi
#
# Usage:
# $ git clone https://github.com/dtrip/rtl8812au.git
# $ cd rtl8812au
# $ patch -p1 -i rtl8812au-rpi.patch
#
# Special thanks to:
# pravinbravi <http://ubuntuforums.org/member.php?u=538656>
# http://ubuntuforums.org/showthread.php?t=2275709
#
# License:
# GPLv2
# <https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html>
#

diff --git a/Makefile b/Makefile
index b79a7d8..013edd6 100644
--- a/Makefile
+++ b/Makefile
@@ -36,7 +36,7 @@ CONFIG_SDIO_HCI = n
 CONFIG_GSPI_HCI = n
 
 CONFIG_MP_INCLUDED = y
-CONFIG_POWER_SAVING = y
+CONFIG_POWER_SAVING = n
 CONFIG_USB_AUTOSUSPEND = n
 CONFIG_HW_PWRP_DETECTION = n
 CONFIG_WIFI_TEST = n
@@ -49,13 +49,13 @@ CONFIG_EXT_CLK = n
 CONFIG_FTP_PROTECT = n
 CONFIG_WOWLAN = n
 
-CONFIG_PLATFORM_I386_PC = y
+CONFIG_PLATFORM_I386_PC = n
 CONFIG_PLATFORM_ANDROID_X86 = n
 CONFIG_PLATFORM_JB_X86 = n
 CONFIG_PLATFORM_ARM_S3C2K4 = n
 CONFIG_PLATFORM_ARM_PXA2XX = n
 CONFIG_PLATFORM_ARM_S3C6K4 = n
-CONFIG_PLATFORM_ARM_RPI = n
+CONFIG_PLATFORM_ARM_RPI = y
 CONFIG_PLATFORM_MIPS_RMI = n
 CONFIG_PLATFORM_RTD2880B = n
 CONFIG_PLATFORM_MIPS_AR9132 = n
diff --git a/os_dep/linux/usb_intf.c b/os_dep/linux/usb_intf.c
index bf5fb69..fdc4a20 100644
--- a/os_dep/linux/usb_intf.c
+++ b/os_dep/linux/usb_intf.c
@@ -286,6 +286,7 @@ static struct usb_device_id rtw_usb_id_tbl[] ={
 	/*=== Customer ID ===*/
 	{USB_DEVICE(0x7392, 0xA811),.driver_info = RTL8821}, /* Edimax - Edimax */
 	{USB_DEVICE(0x7392, 0xA812),.driver_info = RTL8821}, /* Edimax - EW-7811UTC */
+    {USB_DEVICE(0x7392, 0xA813),.driver_info = RTL8821}, /* Amped Wireless UA230A */
 	{USB_DEVICE(0x2001, 0x3314),.driver_info = RTL8821}, /* D-Link - Cameo */
 	{USB_DEVICE(0x2001, 0x3318),.driver_info = RTL8821}, /* D-Link - DWA-172 */
 	{USB_DEVICE(0x0846, 0x9052),.driver_info = RTL8821}, /* Netgear - A6100 */
